### PR TITLE
Added CallbackPositionProperty

### PR DESCRIPTION
**Pull request by jfayot**
# Description

This is a proposal for a `CallbackPositionProperty`, without making TS expose Property types as generics as mentioned [here](https://github.com/CesiumGS/cesium/issues/8944#issuecomment-649890003)

## Issue number and link

- #8944 
- #5534 
- #2597 

## Testing plan

Added specific spec file with 100% coverage on the `CallbackPositionProperty` class.
Updated `PathVizualizer` spec file to allow `CallbackPositionProperty` sub-sampling.
Added a specific sandcastle demo.

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by ggetz** [comment]
Thanks @jfayot! 

We'll review this shortly, likely early next week. If we merge, it will go into the October 1 release.

**Comment by jfayot** [comment]
This is ready :eyes:

**Comment by ggetz** [comment]
Thank you for your patience @jfayot! I will be able to do a full review tomorrow.

File: Apps/Sandcastle/gallery/Callback Position Property.html [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"
+    />
+    <meta
+      name="description"
+      content="Use Viewer to start building new applications or easily embed Cesium into existing applications."
```
Comment (ggetz) [inline Apps/Sandcastle/gallery/Callback Position Property.html:12]: Let's make this more relevant to the example. Something like:

```suggestion
      content="Use a CallbackPositionProperty when your data can't be pre-computed or needs to be derived from other properties at runtime."
```

**Review by ggetz — COMMENTED** [review]
This looks great @jfayot! And this is a fantastic Sandcastle example too!

A few small comments, but otherwise the code is looking great. Thanks again for the PR!

File: Apps/Sandcastle/gallery/Callback Position Property.html [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        scene.morphStart.addEventListener(function () {
+          morphing = true;
+        });
+        scene.morphComplete.addEventListener(function () {
+          morphing = false;
+        });
+
+        // Update camera to follow entity's position and orientation
+        viewer.clock.onTick.addEventListener(function (clock) {
+          if (morphing) {
```
[the remark this answers is not in the document]
Comment (jfayot) [reply Apps/Sandcastle/gallery/Callback Position Property.html]: Yes easier !

File: Apps/Sandcastle/gallery/Callback Position Property.html (hunk 1 above)
Comment (jfayot) [reply Apps/Sandcastle/gallery/Callback Position Property.html]: Yes !

**Comment by jfayot** [comment]
Thanks for the review @ggetz ! Updated according to your comments. Should I resolve conflicts ?

**Comment by jfayot** [comment]
Resolved conflicts, should be ready to merge, but you'd better double check, I'm not a git expert...

**Comment by ggetz** [comment]
Looks great! Thanks again @jfayot!

**Pull request merged**